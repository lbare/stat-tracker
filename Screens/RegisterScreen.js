import React, { useContext, useEffect } from "react";
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  Keyboard,
  KeyboardAvoidingView,
} from "react-native";
import { register } from "../services/firebase";
import { AuthContext } from "../components/AuthContext";

const Register = ({ navigation }) => {
  const [email, setEmail] = React.useState("");
  const [password, setPassword] = React.useState("");
  const [confirmPassword, setConfirmPassword] = React.useState("");

  const user = useContext(AuthContext);

  const onLoginPress = () => navigation.navigate("Login");
  const onRegisterPress = () => {
    if (password !== confirmPassword) {
      alert("Passwords don't match.");
      return;
    }
    register(email, password);
  };

  useEffect(() => {
    if (user) navigation.navigate("Home");
  }, []);

  return (
    <KeyboardAvoidingView
      className='flex-1 justify-center bg-white'
      behavior='padding'
    >
      <View className='h-1/4'>
        <Text className='text-7xl pb-20 font-black text-center text-blue-500 pt-1'>
          Sign Up
        </Text>
      </View>
      <View className='h-1/4'>
        <TextInput
          className='bg-gray-100 border border-gray-500 mx-10 p-4 rounded-lg'
          title='Email'
          placeholder='Email'
          blurOnSubmit={false}
          secureTextEntry={false}
          keyboardType='email-address'
          textContentType='emailAddress'
          value={email}
          onChangeText={(value) => setEmail(value)}
          onSubmitEditing={() => Keyboard.dismiss()}
          returnKeyType='next'
        />
        <TextInput
          className='bg-gray-100 border border-gray-500 mx-10 mt-5 p-4 rounded-lg'
          title='Password'
          placeholder='Password'
          blurOnSubmit={false}
          secureTextEntry={true}
          keyboardType='default'
          textContentType='password'
          value={password}
          onSubmitEditing={() => Keyboard.dismiss()}
          onChangeText={(value) => setPassword(value)}
          returnKeyType='next'
        />
        <TextInput
          className='bg-gray-100 border border-gray-500 mx-10 mt-5 p-4 rounded-lg'
          title='Confirm Password'
          placeholder='Confirm Password'
          blurOnSubmit={true}
          secureTextEntry={true}
          keyboardType='default'
          textContentType='password'
          value={confirmPassword}
          onChangeText={(value) => setConfirmPassword(value)}
          returnKeyType='done'
          onSubmitEditing={onRegisterPress}
        />
        <TouchableOpacity
          className='bg-blue-500 mx-10 mt-5 py-5 rounded-lg'
          onPress={onRegisterPress}
        >
          <Text className='text-xl text-center text-white'>Create Account</Text>
        </TouchableOpacity>
        <Text
          className='mt-5 text-center text-20 text-blue-900 text-sm'
          onPress={onLoginPress}
          suppressHighlighting={true}
        >
          Already have an account?{"\n"}SIGN IN
        </Text>
      </View>
    </KeyboardAvoidingView>
  );
};

export default Register;
