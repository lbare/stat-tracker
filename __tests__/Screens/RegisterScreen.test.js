import React from 'react';
import renderer from 'react-test-renderer';

import RegisterScreen from '../../Screens/RegisterScreen';

describe('<RegisterScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<RegisterScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
