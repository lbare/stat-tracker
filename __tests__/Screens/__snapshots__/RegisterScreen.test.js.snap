// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterScreen /> renders correctly 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "justifyContent": "center",
      },
      {
        "backgroundColor": "#fff",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "height": "25%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 72,
            "lineHeight": 72,
          },
          {
            "paddingBottom": 80,
          },
          {
            "fontWeight": "900",
          },
          {
            "textAlign": "center",
          },
          {
            "color": "#3b82f6",
          },
          {
            "paddingTop": 4,
          },
        ]
      }
    >
      Sign Up
    </Text>
  </View>
  <View
    style={
      [
        {
          "height": "25%",
        },
      ]
    }
  >
    <TextInput
      blurOnSubmit={false}
      keyboardType="email-address"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Email"
      returnKeyType="next"
      secureTextEntry={false}
      style={
        [
          {
            "backgroundColor": "#f3f4f6",
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "borderBottomColor": "#6b7280",
            "borderLeftColor": "#6b7280",
            "borderRightColor": "#6b7280",
            "borderTopColor": "#6b7280",
          },
          {
            "marginLeft": 40,
            "marginRight": 40,
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
          },
        ]
      }
      textContentType="emailAddress"
      title="Email"
      value=""
    />
    <TextInput
      blurOnSubmit={false}
      keyboardType="default"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Password"
      refInner={
        {
          "current": undefined,
        }
      }
      returnKeyType="next"
      secureTextEntry={true}
      style={
        [
          {
            "backgroundColor": "#f3f4f6",
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "borderBottomColor": "#6b7280",
            "borderLeftColor": "#6b7280",
            "borderRightColor": "#6b7280",
            "borderTopColor": "#6b7280",
          },
          {
            "marginLeft": 40,
            "marginRight": 40,
          },
          {
            "marginTop": 20,
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
          },
        ]
      }
      textContentType="password"
      title="Password"
      value=""
    />
    <TextInput
      blurOnSubmit={true}
      keyboardType="default"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Confirm Password"
      refInner={
        {
          "current": undefined,
        }
      }
      returnKeyType="done"
      secureTextEntry={true}
      style={
        [
          {
            "backgroundColor": "#f3f4f6",
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "borderBottomColor": "#6b7280",
            "borderLeftColor": "#6b7280",
            "borderRightColor": "#6b7280",
            "borderTopColor": "#6b7280",
          },
          {
            "marginLeft": 40,
            "marginRight": 40,
          },
          {
            "marginTop": 20,
          },
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
          {
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
          },
        ]
      }
      textContentType="password"
      title="Confirm Password"
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#3b82f6",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "marginLeft": 40,
          "marginRight": 40,
          "marginTop": 20,
          "opacity": 1,
          "paddingBottom": 20,
          "paddingTop": 20,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 20,
              "lineHeight": 28,
            },
            {
              "textAlign": "center",
            },
            {
              "color": "#fff",
            },
          ]
        }
      >
        Create Account
      </Text>
    </View>
    <Text
      onPress={[Function]}
      style={
        [
          {
            "marginTop": 20,
          },
          {
            "textAlign": "center",
          },
          {
            "color": "#1e3a8a",
          },
          {
            "fontSize": 14,
            "lineHeight": 20,
          },
        ]
      }
      suppressHighlighting={true}
    >
      Already have an account?
      

      SIGN IN
    </Text>
  </View>
</View>
`;
